### PR TITLE
Total Mint - Event DB

### PR DESCRIPTION
**Pull request by mohsenno1**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/dbs/event/mint_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import (
+	"os"
+	"testing"
+	"time"
+
+	"0chain.net/chaincore/config"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddMint(t *testing.T) {
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/dbs/event/mint_test.go]: most likely we don't have postgres deployed for test env, so it won't work this way

File: code/go/0chain.net/smartcontract/dbs/event/snapshot.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package event
+
+// swagger:model Block
+type Snapshot struct {
+	Round           int64  `gorm:"primaryKey;autoIncrement:false" json:"round"`
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/dbs/event/snapshot.go:5]: it's not a good idea to make it autoIncremented, in theory we can skip rounds, sharders do not have control over round numbering

File: code/go/0chain.net/smartcontract/dbs/event/snapshot_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		}); err != nil {
+			t.Error(err)
+			return
+		}
+	}
+}
+
+func SetupDatabase(t *testing.T) *EventDb {
+
+	access := config.DbAccess{
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/dbs/event/snapshot_test.go:73]: the same, no db on unit test environment

File: code/go/0chain.net/smartcontract/dbs/event/snapshot.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	BlockHash       string `json:"block_hash"`
+	MintTotalAmount int64  `json:"mint_total_amount"`
+}
+
+func (edb *EventDb) GetRoundsMintTotal(from, to int64) (int64, error) {
+	var total int64
+	return total, edb.Store.Get().Model(&Snapshot{}).Where("round between ? and ?", from, to).Select("sum(mint_total_amount)").Scan(&total).Error
+}
+
+func (edb *EventDb) addOrUpdateTotalMint(mint *Mint) error {
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/dbs/event/snapshot.go:15]: use upsert here https://gorm.io/docs/create.html#Upsert-x2F-On-Conflict

```go
// Update columns to new value on `id` conflict
db.Clauses(clause.OnConflict{
  Columns:   []clause.Column{{Name: "round"}},
  DoUpdates: clause.AssignmentColumns([]string{"mint_total_amount"}),
}).Create(&mint)
```

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/dbs/event/snapshot.go (hunk 2 above)
Comment (mohsenno1) [reply code/go/0chain.net/smartcontract/dbs/event/snapshot.go:5]: This is autoIncrement:false. By default it's auto increment and I set it to false.

File: code/go/0chain.net/smartcontract/dbs/event/snapshot.go (hunk 4 above)
Comment (mohsenno1) [reply code/go/0chain.net/smartcontract/dbs/event/snapshot.go:15]: With gorm 1.9.x or below, it's more effecient to update first, then insert when not existed.
But both doing the same thing.

**Comment by dabasov** [comment]
merged in snapshot PR

**Pull request closed without merging**